### ISSUE TITLE
UI to associate pools and stewards with relevant NFTs

### ISSUE THREAD

**Issue by decentralauren** [GoodDollar/GoodCollective#158]
**Business Requirement**
Each payment for direct payment GoodColllectives must trace back to the NFT that triggered the payment. This should be accessible from both the steward's wallet profile and the collective profile. As NFTs sometimes trigger multiple payments for different wallets, a new design must be created to account for and link to the NFTs Stewards were paid from, not necessarily the NFTs they hold.

(will update how to resolve with clarity from Lewis)

[Slack thread ](https://gooddollar.slack.com/archives/C04QKP36076/p1706874560650899)

**OLD DESIGN**
<img width="719" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM0lqQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--131ea0529d9f5be8c51e031e6c2d93ff542d9487/image.png" alt="image.png" />

**Comment by decentralauren** [comment]
@L03TJ3 Please add in description above any further detail on the "how" once we have it. Will then be in our court to update the design (just the Collectives page, right?)

**Comment by L03TJ3** [comment]
@decentralauren The only thing I see possible is to just link to the transaction.

In the transaction that steward(s) are paid, is the same transaction that the NFT is minted.
The NFT itself does not contain any information about all the stewards.
So on chain it is not registered as being send and nft, or being owner/part of the nft.

Context:
https://celoscan.io/token/0x2af5db8249229ba696e0bac1a382adb1fee040a6
Above you could track an NFT send out, for all pools. it only shows the main owner that is receiving the nft

https://celoscan.io/tx/0x4984175b47d5b387ed7a58dc97982e8a8f9e9d95292b68d33e60cbd2cabf63fd
Here you see a mint-nft transaction involving two stewards.

So from my pov, the tx transaction to link to makes the most sense (we could show it on UI as NFT, but not sure if that is confusing too people)

cc: @sirpy if you have other suggestion?

**Comment by decentralauren** [comment]
@L03TJ3 as just discussed:

Since NFTs may pay out multiple stewards, what we display in the UI for each steward's wallet should be the NFTs that have paid out the steward (rather than the NFTs held by the steward). This is to account for future events in which 1 NFT is for 1 claim done by multiple stewards (in which multiple stewards were paid).

**Comment by L03TJ3** [comment]
Activity page: showing all pools for a steward listing per action links to explorer

**Comment by decentralauren** [comment]
@sirpy ok - but the entire point of the platform is to be able to audit the individual claim that triggered the payment to the steward. I thought we had reference to the IPFS and other vectors in the Silvi claim?

**Comment by sirpy** [comment]
@decentralauren yes we keep it. But we are discussing what is the most user friendly way to actually see this data.

1. the ipfs file is supposed to be the proof of action created by the app, it doesnt necessarily includes the payment info. the ipfs data should be visible on the explorer NFT page considering the app (ie silvi) created it correctly.   

2. the MINT tx, shows the actual payments as distributed by the app. and it also shows the NFT created and links to it.

@L03TJ3 The actual correct way to link to the nft is like so (nft address/nft id)
https://celoscan.io/nft/0x2af5db8249229ba696e0bac1a382adb1fee040a6/18650595618017598607434303353828833723706945133047691375348409019342376399484

@L03TJ3 
If the nft tokenUri links to a valid json file it should show it as properties.
so make sure when you mint an NFT you also link it with a valid json file.

**Comment by L03TJ3** [comment]
@sirpy the whole discussion was not about the link to the nft. the discussion was that if there were multiple stewards associated with it, that the nft page is confusing since its only showing one of the stewards as owner.

**Comment by decentralauren** [comment]
Closing this ticket and replacing with this one - https://github.com/GoodDollar/GoodCollective/issues/221
